tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpMessageHandler.cs

```diff
@@ -22,6 +22,7 @@ internal class MockHttpMessageHandler : HttpMessageHandler
         public IDictionary<string, string> ExpectedQueryParams { get; set; }
         public IDictionary<string, string> ExpectedPostData { get; set; }
         public IDictionary<string, string> ExpectedRequestHeaders { get; set; }
+        public IDictionary<string, string> UnexpectedRequestHeaders { get; set; }
```

Comment:
You don't do anything with the value, just with the key. So this should be a list? 

Alternatively, you can just inspect the `ActualRequestMessage`  in your test and assert directly. This doesn't look like a helper that we'd use too much.

Reply:
Yes. makes sense. just a list